packages/node/src/lib/patch-response.ts

```diff
@@ -2,17 +2,21 @@
 // so we can send it off to the metrics server
 // It's unfortunate that this isn't accessible
 // natively. This may take up lots of memory on
+import type { ExtendedResponse } from './log';
+
 // big responses, we can make it configurable in future
-export function patchResponse(res) {
+export function patchResponse(res: ExtendedResponse) {
   const { write, end } = res;
 
   res._body = '';
 
+  // @ts-expect-error this feels scary to mess with further
```

Comment:
Can you change the comment to clarify what it is you're not sure of and what you'd prefer to do, so a future reader could dive in deeper if necessary?

Reply:
added in 7d42eacab182c6c615069da5dbb63a51efd76ac3